html2canvas@1.0.0-rc.2 makes Angular project build failed,html2canvas@1.0.0-rc.1 is ok.

npm install html2canvas@1.0.0-rc.2 sucessfull ,than call 'ng build'.
In the end, it failed.
I guess it because of html2canvas@1.0.0-rc.2,maybe .

Summary log:
Node#moveTo was deprecated. Use Container#append.
[0mDate: [1m[37m2019-05-29T07:47:36.387Z[39m[22m[0m
[0mHash: [1m[37m3fa4b589eb9668899fde[39m[22m[0m
[0mTime: [1m[37m73413[39m[22mms[0m
[0mchunk {[1m[33mscripts[39m[22m} [1m[32mscripts.78f0bdbed29811d920e0.bundle.js[39m[22m (scripts) 2.06 MB [1m[33m[initial][39m[22m[1m[32m [rendered][39m[22m[0m
[0mchunk {[1m[33m0[39m[22m} [1m[32mstyles.8855f6bd28d07ac6a4de.bundle.css[39m[22m (styles) 485 kB [1m[33m[initial][39m[22m[1m[32m [rendered][39m[22m[0m
[0mchunk {[1m[33m1[39m[22m} [1m[32mpolyfills.3bc34265385d52184eab.bundle.js[39m[22m (polyfills) 86 bytes [1m[33m[initial][39m[22m[1m[32m [rendered][39m[22m[0m
[0mchunk {[1m[33m2[39m[22m} [1m[32mmain.e402deade8b026b7d50e.bundle.js[39m[22m (main) 84 bytes [1m[33m[initial][39m[22m[1m[32m [rendered][39m[22m[0m
[0mchunk {[1m[33m3[39m[22m} [1m[32minline.5bdc54bbc6535e2d60f2.bundle.js[39m[22m (inline) 1.45 kB [1m[33m[entry][39m[22m[1m[32m [rendered][39m[22m[0m
[0m[0m
[0m[1m[31mERROR in node_modules/html2canvas/dist/types/css/index.d.ts(43,21): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(45,22): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(46,23): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(47,25): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(48,23): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(49,21): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(54,26): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(55,27): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(56,30): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(57,29): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(58,21): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(59,23): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(60,24): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(61,22): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(62,21): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(63,23): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(64,24): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(65,22): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(66,16): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(68,14): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(69,12): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(70,17): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(72,16): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(73,18): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(74,17): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(75,20): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(76,16): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(78,21): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(79,24): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(80,20): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(85,14): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(88,19): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(93,15): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(94,16): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(96,25): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(97,17): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(98,20): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(99,16): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(100,22): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(101,17): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(102,16): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(103,13): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(114,14): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(115,13): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(119,23): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m
[0m[1m[31mnode_modules/html2canvas/dist/types/css/index.d.ts(120,19): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m

### Specifications:

 * html2canvas version tested with:
 * Browser & version:1.0.0-rc.2
 * Operating system:Linux

Getting Error - ERROR in node_modules/html2canvas/dist/types/css/index.d.ts(43,21): error TS2304: Cannot find name 'ReturnType'.[39m[22m[0m.
i am using hml2pdf and that is using html2canvas as dependencies .

--- a/src/core/cache-storage.ts
+++ b/src/core/cache-storage.ts
@@ -61,7 +61,7 @@ export class CacheStorage {
     }
 }
 
-interface ResourceOptions {
+export interface ResourceOptions {
     imageTimeout: number;
     useCORS: boolean;
     allowTaint: boolean;

--- a/src/index.ts
+++ b/src/index.ts
@@ -4,25 +4,22 @@ import {Parser} from './css/syntax/parser';
 import {CloneOptions, DocumentCloner} from './dom/document-cloner';
 import {isBodyElement, isHTMLElement, parseTree} from './dom/node-parser';
 import {Logger} from './core/logger';
-import {CacheStorage} from './core/cache-storage';
+import {CacheStorage, ResourceOptions} from './core/cache-storage';
 import {CanvasRenderer, RenderOptions} from './render/canvas/canvas-renderer';
 import {ForeignObjectRenderer} from './render/canvas/foreignobject-renderer';
 
 export type Options = CloneOptions &
-    RenderOptions & {
-        allowTaint: boolean;
+    RenderOptions &
+    ResourceOptions & {
         backgroundColor: string;
         foreignObjectRendering: boolean;
-        imageTimeout: number;
         logging: boolean;
-        proxy?: string;
         removeContainer?: boolean;
-        useCORS: boolean;
     };
 
 const parseColor = (value: string): Color => color.parse(Parser.create(value).parseComponentValue());
 
-const html2canvas = (element: HTMLElement, options: Options): Promise<HTMLCanvasElement> => {
+const html2canvas = (element: HTMLElement, options: Partial<Options> = {}): Promise<HTMLCanvasElement> => {
     return renderElement(element, options);
 };
 
@@ -30,7 +27,7 @@ export default html2canvas;
 
 CacheStorage.setContext(window);
 
-const renderElement = async (element: HTMLElement, opts: Options): Promise<HTMLCanvasElement> => {
+const renderElement = async (element: HTMLElement, opts: Partial<Options>): Promise<HTMLCanvasElement> => {
     const ownerDocument = element.ownerDocument;
 
     if (!ownerDocument) {
@@ -43,50 +40,43 @@ const renderElement = async (element: HTMLElement, opts: Options): Promise<HTMLC
         throw new Error(`Document is not attached to a Window`);
     }
 
-    const defaultOptions = {
+    const instanceName = (Math.round(Math.random() * 1000) + Date.now()).toString(16);
+
+    const {width, height, left, top} =
+        isBodyElement(element) || isHTMLElement(element) ? parseDocumentSize(ownerDocument) : parseBounds(element);
+
+    const defaultResourceOptions = {
         allowTaint: false,
-        backgroundColor: '#ffffff',
         imageTimeout: 15000,
-        logging: true,
         proxy: undefined,
+        useCORS: false
+    };
+
+    const resourceOptions: ResourceOptions = {...defaultResourceOptions, ...opts};
+
+    const defaultOptions = {
+        backgroundColor: '#ffffff',
+        cache: opts.cache ? opts.cache : CacheStorage.create(instanceName, resourceOptions),
+        logging: true,
         removeContainer: true,
         foreignObjectRendering: false,
         scale: defaultView.devicePixelRatio || 1,
-        useCORS: false,
         windowWidth: defaultView.innerWidth,
         windowHeight: defaultView.innerHeight,
         scrollX: defaultView.pageXOffset,
-        scrollY: defaultView.pageYOffset
+        scrollY: defaultView.pageYOffset,
+        x: left,
+        y: top,
+        width: Math.ceil(width),
+        height: Math.ceil(height),
+        id: instanceName
     };
 
-    const options: Options = {...defaultOptions, ...opts};
+    const options: Options = {...defaultOptions, ...resourceOptions, ...opts};
 
     const windowBounds = new Bounds(options.scrollX, options.scrollY, options.windowWidth, options.windowHeight);
 
-    // http://www.w3.org/TR/css3-background/#special-backgrounds
-    const documentBackgroundColor = ownerDocument.documentElement
-        ? parseColor(getComputedStyle(ownerDocument.documentElement).backgroundColor as string)
-        : COLORS.TRANSPARENT;
-    const bodyBackgroundColor = ownerDocument.body
-        ? parseColor(getComputedStyle(ownerDocument.body).backgroundColor as string)
-        : COLORS.TRANSPARENT;
-
-    const backgroundColor =
-        element === ownerDocument.documentElement
-            ? isTransparent(documentBackgroundColor)
-                ? isTransparent(bodyBackgroundColor)
-                    ? options.backgroundColor
-                        ? parseColor(options.backgroundColor)
-                        : null
-                    : bodyBackgroundColor
-                : documentBackgroundColor
-            : options.backgroundColor
-            ? parseColor(options.backgroundColor)
-            : null;
-
-    const instanceName = (Math.round(Math.random() * 1000) + Date.now()).toString(16);
     Logger.create(instanceName);
-    const cache = CacheStorage.create(instanceName, options);
     Logger.getInstance(instanceName).debug(`Starting document clone`);
     const documentCloner = new DocumentCloner(element, {
         id: instanceName,
@@ -102,22 +92,37 @@ const renderElement = async (element: HTMLElement, opts: Options): Promise<HTMLC
 
     const container = await documentCloner.toIFrame(ownerDocument, windowBounds);
 
-    const {width, height, left, top} =
-        isBodyElement(clonedElement) || isHTMLElement(clonedElement)
-            ? parseDocumentSize(ownerDocument)
-            : parseBounds(clonedElement);
+    // http://www.w3.org/TR/css3-background/#special-backgrounds
+    const documentBackgroundColor = ownerDocument.documentElement
+        ? parseColor(getComputedStyle(ownerDocument.documentElement).backgroundColor as string)
+        : COLORS.TRANSPARENT;
+    const bodyBackgroundColor = ownerDocument.body
+        ? parseColor(getComputedStyle(ownerDocument.body).backgroundColor as string)
+        : COLORS.TRANSPARENT;
+
+    const bgColor = opts.backgroundColor;
+    const defaultBackgroundColor = typeof bgColor === 'string' ? parseColor(bgColor) : 0xffffffff;
+
+    const backgroundColor =
+        element === ownerDocument.documentElement
+            ? isTransparent(documentBackgroundColor)
+                ? isTransparent(bodyBackgroundColor)
+                    ? defaultBackgroundColor
+                    : bodyBackgroundColor
+                : documentBackgroundColor
+            : defaultBackgroundColor;
 
     const renderOptions = {
         id: instanceName,
-        cache,
+        cache: options.cache,
         backgroundColor,
         scale: options.scale,
-        x: typeof options.x === 'number' ? options.x : left,
-        y: typeof options.y === 'number' ? options.y : top,
+        x: options.x,
+        y: options.y,
         scrollX: options.scrollX,
         scrollY: options.scrollY,
-        width: typeof options.width === 'number' ? options.width : Math.ceil(width),
-        height: typeof options.height === 'number' ? options.height : Math.ceil(height),
+        width: options.width,
+        height: options.height,
         windowWidth: options.windowWidth,
         windowHeight: options.windowHeight
     };
@@ -131,7 +136,7 @@ const renderElement = async (element: HTMLElement, opts: Options): Promise<HTMLC
     } else {
         Logger.getInstance(instanceName).debug(`Document cloned, using computed rendering`);
 
-        CacheStorage.attachInstance(cache);
+        CacheStorage.attachInstance(options.cache);
         Logger.getInstance(instanceName).debug(`Starting DOM parsing`);
         const root = parseTree(clonedElement);
         CacheStorage.detachInstance();

--- a/src/render/canvas/canvas-renderer.ts
+++ b/src/render/canvas/canvas-renderer.ts
@@ -39,11 +39,14 @@ import {SelectElementContainer} from '../../dom/elements/select-element-containe
 import {IFrameElementContainer} from '../../dom/replaced-elements/iframe-element-container';
 import {TextShadow} from '../../css/property-descriptors/text-shadow';
 
+export type RenderConfigurations = RenderOptions & {
+    backgroundColor: Color | null;
+};
+
 export interface RenderOptions {
     id: string;
     scale: number;
     canvas?: HTMLCanvasElement;
-    backgroundColor: Color | null;
     x: number;
     y: number;
     scrollX: number;
@@ -60,11 +63,11 @@ const MASK_OFFSET = 10000;
 export class CanvasRenderer {
     canvas: HTMLCanvasElement;
     ctx: CanvasRenderingContext2D;
-    options: RenderOptions;
+    options: RenderConfigurations;
     private readonly _activeEffects: IElementEffect[] = [];
     private readonly fontMetrics: FontMetrics;
 
-    constructor(options: RenderOptions) {
+    constructor(options: RenderConfigurations) {
         this.canvas = options.canvas ? options.canvas : document.createElement('canvas');
         this.ctx = this.canvas.getContext('2d') as CanvasRenderingContext2D;
         this.options = options;

--- a/src/render/canvas/foreignobject-renderer.ts
+++ b/src/render/canvas/foreignobject-renderer.ts
@@ -1,4 +1,4 @@
-import {RenderOptions} from './canvas-renderer';
+import {RenderConfigurations} from './canvas-renderer';
 import {Logger} from '../../core/logger';
 import {createForeignObjectSVG} from '../../core/features';
 import {asString} from '../../css/types/color';
@@ -6,9 +6,9 @@ import {asString} from '../../css/types/color';
 export class ForeignObjectRenderer {
     canvas: HTMLCanvasElement;
     ctx: CanvasRenderingContext2D;
-    options: RenderOptions;
+    options: RenderConfigurations;
 
-    constructor(options: RenderOptions) {
+    constructor(options: RenderConfigurations) {
         this.canvas = options.canvas ? options.canvas : document.createElement('canvas');
         this.ctx = this.canvas.getContext('2d') as CanvasRenderingContext2D;
         this.options = options;

